### PR TITLE
Site Settings: Adds holidaysnow to general site settings

### LINKED ISSUE #1123: Holiday Snow settings: add site and user option
It's December 1st! Snow is back on WordPress.com and Jetpack sites, and will fall until January 4th.

There are 2 settings related to that feature:
1. A site setting, available under Settings > General in wp-admin. I didn't find that option in the API when looking at the `sites/%ID` endpoint.
2. A user setting, that used to be available in the account settings, but that wasn't ported to Calypso. It allows WordPress.com users to disable snow for all WordPress.com blogs, when they're logged in. The option is available in the API, via the `/me/settings/` endpoint (option name is `holidaysun`).

It would be nice if both options were made available in Calypso.

### PR DESCRIPTION
**Pull request by ebinnion**
Closes #1123

@jeherve reported in #1123 that the holiday snow options weren't ported to Calypso. In #1138, I added the holiday snow override to `/me` and this PR adds the holiday snow site setting to the `/settings/general/$site`.

__Note:__ The first two commits are ESLint fixes. So, if you'd like to see the changeset to add `holidaysnow`, see the third commit.

To test:
- Checkout `add/site-settings-holiday-snow` branch
- Go to `/me/account` and ensure you have holiday snow enabled
- Go to `/settings/general/$site` where `$site` is a WP.com site and ensure that snow is enabled
- Go to `$site`. Do you see snow? Good
- Go to `/settings/general/$site` and disable snow.
- Go to `$site`. You should not see snow.
- Go to `/settings/general/$site` where `$site` is a Jetpack site. You should not see the holiday snow option yet. Since this requires an API change, it is unlikely that holiday snow will be changeable via the API until 4.0 (late January 2016)

cc @lancewillett who pointed the issue out to me and @lezama who has done some work in `form-general`.

__WP.com site__
screen shot 1

__Jetpack site__
screen shot

### CODE REVIEW

**Comment by lancewillett** [comment]
QA :+1: — works exactly as expected. Nice!

File: client/my-sites/site-settings/form-general.jsx [hunk 1]
```diff
@@ -184,24 +193,7 @@ module.exports = React.createClass( {
...
-						value="-1"
-						checked={ - 1 === parseInt( this.state.blog_public, 10 ) }
-						onChange={ this.handleRadio }
-						disabled={ this.state.fetchingSettings }
-						onClick={ this.recordEvent.bind( this, 'Clicked Site Visibility Radio Button' ) }
-					/>
-					<span>{ this.translate( 'I would like my site to be private, visible only to users I choose' ) }</span>
-				</label>;
-		}
+		var site = this.props.site;
```
Comment (lezama) [inline client/my-sites/site-settings/form-general.jsx]: this could be `const`

File: client/my-sites/site-settings/form-general.jsx [hunk 2]
```diff
@@ -237,7 +229,7 @@ module.exports = React.createClass( {
 					</p>
 				</label>
 
-				{ ! site.jetpack ?
+				{ ! site.jetpack &&
```
Comment (lezama) [inline client/my-sites/site-settings/form-general.jsx]: :heart:

File: client/my-sites/site-settings/form-general.jsx [hunk 3]
```diff
@@ -113,20 +118,24 @@ module.exports = React.createClass( {
 		}
 
 		if ( config.isEnabled( 'upgrades/domain-search' ) ) {
-			customAddress = <a
-								href={ '/domains/add/' + site.slug  }
-								className="button"
-								onClick={ this.trackUpgradeClick }>{ this.translate( 'Add a custom address', { context: 'Site address, domain' } ) }
-							</a>;
+			customAddress = (
+				<a
```
Comment (lezama) [inline client/my-sites/site-settings/form-general.jsx]: this could be a `'components/button'`? probably not related to this PR :)

**Comment by lezama** [comment]
:shipit:

**Comment by alisterscott** [comment]
I can no longer see the option to disable snow for a simple site in Calypso - it still exists in wp-admin for simple sites. Raised here: https://github.com/Automattic/wp-calypso/issues/20395

**Pull request merged**